src/project.cc

```diff
@@ -825,7 +828,7 @@ TEST_SUITE("Project") {
          "debian_jessie_amd64-sysroot",
          "-fno-exceptions",
          "-fvisibility-inlines-hidden",
-         "-xc++",
+         "../../ash/login/ui/lock_screen_sanity_unittest.cc",
```

Comment:
This path should probably be normalized (ie, it should have & before it).

Reply:
What is the `&` for?

Reply:
It means that the path was normalized. Because it is a test normalization doesn't actually happen, just & is prepended.

Reply:
ie, the path should be `"&../../ash/login/ui/lock_screen_sanity_unittest.cc"`

Reply:
Does clang USR encode absolute filename? If not, is it fine to leave filename arguments as is?

Reply:
The main source filename is returned by `clang_CompileCommand_getFilename(cx_command)`. It is difficult to recognize the main source filename (thinking of `-include -isystem -I`) and put a `&` before it to mark it as normalized (for testing purpose).

Reply:
This still fails because the path is not being normalized.  The filename should be in `entry.file`.

One alternate approach that does not depend on filename (though I'd prefer checking the filename):
```
if (arg[0] != '-')
  arg = cleanup_maybe_relative_path(arg);
```

Reply:
Rebased.  Added these two lines.

Reply:
After switching to `-working-directory`, this normalization is removed.